src/configuration/configuration_manager.cpp

```diff
@@ -0,0 +1,164 @@
+//
+// Identification: src/configuration/configuration_manager.cpp
+//
+// Copyright (c) 2015-2017, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+#define __PELOTON_CONFIG__
+
+#include "common/exception.h"
+#include "concurrency/transaction_manager_factory.h"
+#include "type/ephemeral_pool.h"
+#include "configuration/configuration_manager.h"
+#include "configuration/configuration.h"
+#include "catalog/config_catalog.h"
+
+namespace peloton {
+namespace configuration {
+
+uint64_t Config::GET_INT(const std::string& name) {
+  return peloton::configuration::ConfigurationManager::GetInstance()->GetValue<uint64_t>(name);
+}
+
+bool Config::GET_BOOL(const std::string& name) {
+  return peloton::configuration::ConfigurationManager::GetInstance()->GetValue<bool>(name);
+}
+
+std::string Config::GET_STRING(const std::string& name) {
+  return peloton::configuration::ConfigurationManager::GetInstance()->GetValue<std::string>(name);
+}
+
+void Config::SET_INT(const std::string &name, uint64_t value) {
+  peloton::configuration::ConfigurationManager::GetInstance()->SetValue<uint64_t>(name, value);
+}
+
+void Config::SET_BOOL(const std::string &name, bool value) {
+  peloton::configuration::ConfigurationManager::GetInstance()->SetValue<bool>(name, value);
+}
+
+void Config::SET_STRING(const std::string &name, const std::string &value) {
+  peloton::configuration::ConfigurationManager::GetInstance()->SetValue<std::string>(name, value);
+}
+
+void Config::init_parameters() {
+  drop_parameters();
+  register_parameters();
+}
+
+void Config::drop_parameters() {
+  auto config_manager = ConfigurationManager::GetInstance();
+  config_manager->Clear();
+}
+
+ConfigurationManager* ConfigurationManager::GetInstance() {
+  static std::unique_ptr<ConfigurationManager> config_manager(
+          new ConfigurationManager());
+  return config_manager.get();
+}
+
+void ConfigurationManager::InitializeCatalog() {
+  auto config_catalog = peloton::catalog::ConfigCatalog::GetInstance();
+
+  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+  auto txn = txn_manager.BeginTransaction();
+  type::AbstractPool *pool = pool_.get();
+
+  for (auto conf : config) {
+    config_catalog->DeleteConfig(conf.first, txn);
+    config_catalog->InsertConfig(conf.first, to_string(conf.second.value, conf.second.value_type),
+                                 conf.second.value_type,
+                                 conf.second.desc, "", "", conf.second.default_value,
+                                 conf.second.is_mutable, conf.second.is_persistent,
+                                 pool, txn);
+  }
+  txn_manager.CommitTransaction(txn);
+  catalog_initialized = true;
+}
+
+const std::string ConfigurationManager::GetInfo() const {
+  std::stringstream str;
+  char buffer[256];
+  sprintf(buffer, "%30s", "//===-------------- PELOTON CONFIGURATION --------------===//");
```

Comment:
Don't use `sprintf`. Use `std::ostringstream`. See the other `GetInfo()` implementations.